test/seqan3-test.cmake

```diff
@@ -117,9 +120,7 @@ macro (seqan3_require_benchmark)
     add_library (benchmark STATIC IMPORTED)
     add_dependencies (benchmark google_benchmark)
     set_target_properties (benchmark PROPERTIES IMPORTED_LOCATION "${benchmark_library}")
-
-    list (APPEND SEQAN3_BENCHMARK_LIBRARIES "benchmark")
-    list (APPEND SEQAN3_BENCHMARK_LIBRARIES "pthread")
+    set_property (TARGET benchmark APPEND PROPERTY INTERFACE_LINK_LIBRARIES "pthread")
```

Comment:
Why is this required? We have:
`target_link_libraries (seqan3_test INTERFACE "pthread")`
and
`target_link_libraries (seqan3_performance_test INTERFACE "benchmark")`

Reply:
google benchmark itself needs pthread. If you don't specify it here it will be at the wrong position in the positions of the flags and the benchmark library might not know some symbols.

We can drop `target_link_libraries (seqan3_test INTERFACE "pthread")`, because it is at the wrong place/position.